django/utils/http.py

```diff
@@ -310,46 +311,31 @@ def escape_leading_slashes(url):
     return url
 
 
-def _parseparam(s):
-    while s[:1] == ";":
-        s = s[1:]
-        end = s.find(";")
-        while end > 0 and (s.count('"', 0, end) - s.count('\\"', 0, end)) % 2:
-            end = s.find(";", end + 1)
-        if end < 0:
-            end = len(s)
-        f = s[:end]
-        yield f.strip()
-        s = s[end:]
+def _cgi_compatible_params(params, limit):
+    pdict = {}
+    key = params.pop(0)[0].strip().lower()
+    params = list(filter(lambda x: x[0], params))[:limit]
+    for name, value in params:
+        if isinstance(value, tuple):
+            encoding, lang, value = value
+            # email.message params always unquoted with "latin-1"
+            if encoding:
+                value = unquote(quote(value, encoding="latin-1"), encoding=encoding)
```

Comment:
I was reading the documentation for Message.get_param, and I noticed that it shows:

> If your application doesn't care whether the parameter was RFC 2231
> encoded, it can turn the return value into a string as follows:
> ...
> param = email.utils.collapse_rfc2231_value(rawparam)

I tried replacing this if with a call like this:
```suggestion
        value = collapse_rfc2231_value(value)
```

No test failed and in fact the values are properly decoded. Do you see a reason not to use this instead of doing the manual quote/unquote calls?